docs/developer/release-team.md

```diff
@@ -0,0 +1,70 @@
+# Cluster API Release Team
+
+## Overview
+
+In the past, releasing Cluster API has been mostly ad-hoc and relied on one or more contributors to do most of the chore work necessary to prepare the release. One of the major downsides of this approach is that it is often difficult for users and providers to plan around Cluster API releases as they often have little visibility around when a release will happen. 
+
+This document introduces the concept of a release team with the following goals and non-goals:
+
+### Goals
+
+- To improve communication to end users and contributors about CAPI release cadence and target release dates
+- To spread the load of release tasks and involve a bigger, more diverse set of CAPI contributors in cutting releases
+- To look at the Kubernetes SIG release team processes for guidance on releasing best practices
+- To improve tooling, documentation, and automation of the CAPI release process
+
+### Non-Goals/Future work
+
+- To increase the frequency of releases (AKA release cadence). This will be revisited in the future once the release process stabilizes.
+- To change the current proposal (CAEP) process
+- To copy implement all steps of the Kubernetes release process for CAPI releases
+
+Note that this document is intended to be a starting point for the release team. It is not a complete release process document.
+
+More details on the CAPI release process can be found in [this past issue tracking release tasks](https://github.com/kubernetes-sigs/cluster-api/issues/6615) and the current [release documentation](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/developer/releasing.md).
+
+## Duration of Term
+
+Each release team term will last for a quarter, to align with one minor release cycle. A minor release cycle starts right after a minor release and concludes with the release of the next minor release.
+
+As noted above, making changes to  the CAPI release cadence is out of scope for this initial release team process. 
+
+## Specific Responsibilities
+
+- Release patch releases monthly or more often as needed, so users will get fixes & updated dependencies with CVE fix on a predictable cadence
+- Release quarterly (to be revised, see future work) minor releases, so users will get new features on a predictable cadence
+- Monitor CI signal, so a release can be cut at any time, and add CI signal for each new release branch
```

Comment:
I am not well aware of the CAPI process yet and it might be that is already the case, but maybe should also add a note about "Security" releases (for all unplanned or embargoed releases including critical CVEs) or in later iterations of this document if turns out to be necessary? if so, adding the actions and responsible person for that release (AFAIU it should be release lead?) can fit in this list.

Reply:
That's a good question. AFAIK so far we haven't had to deal with embargoed releases / critical CVEs and I don't know if we have an established process, @vincepri @fabriziopandini @sbueringer thoughts?

Reply:
I also don't remember us dealing with embargoed releases / critical CVEs.
Due to this and the fact that we are just starting with this process, I personally think that for now we can assume to treat everything as a "simple" patch, conscious that the attention to communication and scheduling should be proportional to the severity of the issues we are dealing with.